apps/ledger-live-desktop/src/renderer/modals/EditTransaction/index.jsx

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import React, { useCallback, useState } from "react";
+import Modal from "~/renderer/components/Modal";
+import Body from "./Body";
+import type { StepId } from "./types";
+
+type Props = {
+  stepId: StepId,
+  onClose: Function,
+};
+
+const SendModal = ({ stepId: initialStepId, onClose }: Props) => {
+  const [stepId, setStep] = useState(() => initialStepId || "method");
+  const [isNftSend, setIsNFTSend] = useState(false);
+  const handleReset = useCallback(() => setStep("method"), []);
+  const handleStepChange = useCallback(stepId => setStep(stepId), []);
+  const handleSetIsNFTSend = useCallback(isNftSend => setIsNFTSend(isNftSend), []);
```

Comment:
I don't really see the point of use using `useCallback` only to call a `setXXX` function

Use callback caches the returned value at each render and the useState inital a state at each render so the `useCallback` here is kinda useless

Reply:
I know it feels useless, but it keeps the same reference at each render. So it means the `<Body>` component (if memoized correctly) will at a new render compare props and decide if it need a re-render as well. If you're using a simple arrow function, that prop will always be different and forcing a re-render, if you're using a useCallback, reference is preserved, so no re-render there. 

That's how shallow compare is working, and how `memo` is trying to avoid re-rendering.